tests/monitors/kafka/kafka_test.py

```diff
@@ -28,22 +28,73 @@ def run_kafka(version="1.0.1"):
         assert wait_for(p(tcp_socket_open, zkhost, 2181), 60), "zookeeper didn't start"
         with run_service(
             "kafka",
-            environment={"KAFKA_ZOOKEEPER_CONNECT": "%s:2181" % (zkhost,)},
+            environment={"JMX_PORT": "7099", "KAFKA_ZOOKEEPER_CONNECT": "%s:2181" % (zkhost,),
+                         "START_AS": "broker"},
             buildargs={"KAFKA_VERSION": version}
         ) as kafka_container:
+            run_service(
+                "kafka",
+                environment={"START_AS": "create-topic", "KAFKA_ZOOKEEPER_CONNECT": "%s:2181" % (zkhost,)},
+                buildargs={"KAFKA_VERSION": version}
+            )
             yield kafka_container
 
+versions = ["0.9.0.0", "0.10.0", "0.11.0", "1.0.0", "1.0.1", "1.1.1"]
 
-def test_kafka_monitor():
-    with run_kafka() as kafka:
-        with run_agent(textwrap.dedent("""
-        monitors:
-         - type: collectd/kafka
-           host: {0}
-           port: 1099
-        """.format(container_ip(kafka)))) as [backend, _, _]:
-            assert wait_for(p(has_datapoint_with_metric_name, backend, "gauge.kafka-active-controllers")), \
-                    "Didn't get kafka datapoints"
+def test_omitting_kafka_metrics():
+    for version in versions:
+        with run_kafka(version) as kafka:
+            kafkahost = container_ip(kafka)
+            with run_agent(textwrap.dedent("""
+            monitors:
+             - type: collectd/kafka
+               host: {0}
+               port: 7099
+               clusterName: testCluster
+               mBeansToOmit:
+                 - kafka-active-controllers
+            """.format(kafkahost))) as [backend, _, _]:
+                assert not wait_for(p(has_datapoint_with_metric_name, backend, "gauge.kafka-active-controllers")), \
+                        "Didn't get kafka datapoints"
+
+
+def test_all_kafka_monitors():
+    for version in versions:
```

Comment:
This should be done with pytest parameterization instead so that the tests can run in parallel.  See for example https://github.com/signalfx/signalfx-agent/blob/master/tests/monitors/hadoop/hadoop_test.py#L40